### ISSUE TITLE
patience parameter in EarlyStopping and ReduceLROnPlateau interpreted differently

### ISSUE THREAD

**Issue by tbsflk** [keras-team/keras#9541]
The callbacks `EarlyStopping `and `ReduceLROnPlateau` both have a patience parameter which one would expect - also according to the documentation - to have a similar effect. 

However, with the current implementation, a value of patience=2 for EarlyStopping will cause the training to stop when the monitored metric has not improved for the second epoch, while patience=2 for ReduceLROnPlateau will reduce the learning rate after the third epoch without improvement.

I think the reason is that the internal counter `self.wait` is incremented before the check against `patience` in [EarlyStopping ](https://github.com/keras-team/keras/blob/master/keras/callbacks.py#L533 ), but after the check in [ReduceLROnPlateau](ReduceLROnPlateau ) .

This behaviour should be consistent across the different callbacks.

**Comment by sotlampr** [comment]
The same with `EarlyStopping`'s `min_delta` and `ReduceLROnPlateau`'s `epsilon` parameters, they refer to the same thing but have different names.

**Comment by sotlampr** [comment]
Yesss
